inlong-tubemq/tubemq-server/src/main/java/org/apache/inlong/tubemq/server/master/web/MasterStatusCheckFilter.java

```diff
@@ -65,7 +69,10 @@ public void doFilter(ServletRequest request,
             if (TStringUtils.isNotBlank(req.getQueryString())) {
                 sBuilder.append("?").append(req.getQueryString());
             }
-            resp.sendRedirect(sBuilder.toString());
+            this.redirectWhiteList = sBuilder.toString();
+            if (redirectWhiteList.equals(sBuilder.toString())) {
```

Comment:
The `redirectWhiteList.equals(sBuilder.toString()` is always true, so why compare them here?

Reply:
Thanks for your review, In order to fix this warning, I saw the official recommended good case, which needs to be compared with the white list, but I didn't think of a better method.
https://codeql.github.com/codeql-query-help/java/java-xss/

Reply:
In this case, there may be a hidden danger of concurrent requests using global variables.

Reply:
I think so. Do I put redirectWhiteList in the function?

Reply:
But if I write code like this to solve this warning, maybe it's not necessary?

Reply:
refer to the following example
```
public class UrlRedirect extends HttpServlet {
	private static final String VALID_REDIRECT = "http://cwe.mitre.org/data/definitions/601.html";

	protected void doGet(HttpServletRequest request, HttpServletResponse response)
	throws ServletException, IOException {
		// BAD: a request parameter is incorporated without validation into a URL redirect
		response.sendRedirect(request.getParameter("target"));

		// GOOD: the request parameter is validated against a known fixed string
		if (VALID_REDIRECT.equals(request.getParameter("target"))) {
			response.sendRedirect(VALID_REDIRECT);
		}
	}
}
```
I think it's better to add an expected URL rule to match the `sBuilder.toString()` to prevent unexpected values.

Reply:
Let me use regular expressions for matching.